doc/guides/maintaining-the-build-files.md

```diff
@@ -0,0 +1,386 @@
+                        alternative lib name to link to [default: uv]
+    --shared-libuv-libpath=SHARED_LIBUV_LIBPATH
+                        a directory to search for the shared libuv DLL
+    --shared-openssl    link to a shared OpenSSl DLL instead of static linking
+    --shared-openssl-includes=SHARED_OPENSSL_INCLUDES
+                        directory containing OpenSSL header files
+    --shared-openssl-libname=SHARED_OPENSSL_LIBNAME
+                        alternative lib name to link to [default: crypto,ssl]
+    --shared-openssl-libpath=SHARED_OPENSSL_LIBPATH
+                        a directory to search for the shared OpenSSL DLLs
+    --shared-zlib       link to a shared zlib DLL instead of static linking
+    --shared-zlib-includes=SHARED_ZLIB_INCLUDES
+                        directory containing zlib header files
+    --shared-zlib-libname=SHARED_ZLIB_LIBNAME
+                        alternative lib name to link to [default: z]
+    --shared-zlib-libpath=SHARED_ZLIB_LIBPATH
+                        a directory to search for the shared zlib DLL
+    --shared-cares      link to a shared cares DLL instead of static linking
+    --shared-cares-includes=SHARED_LIBCARES_INCLUDES
+                        directory containing cares header files
+    --shared-cares-libname=SHARED_LIBCARES_LIBNAME
+                        alternative lib name to link to [default: cares]
+    --shared-cares-libpath=SHARED_LIBCARES_LIBPATH
+                        a directory to search for the shared cares DLL
+
+  Internationalization:
+    Flags that lets you enable i18n features in Node.js as well as which
+    library you want to build against.
+
+    --with-intl=WITH_INTL
+                        Intl mode (valid choices: none, small-icu, full-icu,
+                        system-icu) [default: small-icu]
+    --without-intl      Disable Intl, same as --with-intl=none (disables
+                        inspector)
+    --with-icu-path=WITH_ICU_PATH
+                        Path to icu.gyp (ICU i18n, Chromium version only.)
+    --with-icu-locales=WITH_ICU_LOCALES
+                        Comma-separated list of locales for "small-icu".
+                        "root" is assumed. [default: root,en]
+    --with-icu-source=WITH_ICU_SOURCE
+                        Intl mode: optional local path to icu/ dir, or
+                        path/URL of icu source archive.
+    --download=DOWNLOAD_LIST
+                        Select which packages may be auto-downloaded. valid
+                        values are: none, all, icu. (default is "none").
+    --download-path=DOWNLOAD_PATH
+                        Download directory [default: ./deps]
+
+  HTTP2:
+    Flags that allows you to control HTTP2 features in Node.js
+
+    --debug-http2       build with http2 debug statements on (default is
+                        false)
+    --debug-nghttp2     build nghttp2 with DEBUGBUILD (default is false)
+```
+
+## Makefile
```

Comment:
I was compiling Alacritty the other day, and I noticed that their `make` actually defaults to showing help (alias for `make help`) rather than doing anything. It actually looks really nice, and it'd be really useful if we could do something similar.

I think a help option to make would be less likely to bitrot than separate documentation.

https://github.com/jwilm/alacritty/blob/245a80078180acc2a0a1addc569c15b77991c1c3/Makefile#L18-L21